src/Components/Product.js

```diff
@@ -5,86 +5,98 @@ import { Link } from "react-router-dom";
 import { useFilter } from "../context/filter-provider";
 import { useWishlist } from "../context/wishlist-provider";
 import { useCart } from "../context/cart-provider";
```

Comment:
You can make a separate file to export functions from contexts and then import those from that single file